wgengine/magicsock/magicsock.go

```diff
@@ -918,6 +918,13 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 			addAddr(netip.AddrPortFrom(v4Addrs[0].Addr(), uint16(port)), tailcfg.EndpointSTUN4LocalPort)
 		}
 	}
+	if s := envknob.String("TS_DEBUG_PRETENDPOINT"); s != "" {
```

Comment:
PRETENDPOINT? Not sure if typo or deliberate

Reply:
very much deliberate :)